pilot/pkg/bootstrap/server.go

```diff
@@ -211,13 +264,26 @@ func (s *Server) Start(stop <-chan struct{}) error {
 
 // initKubeClient creates the k8s client if running in an k8s environment.
 func (s *Server) initKubeClient(args *PilotArgs) error {
-	if hasKubeRegistry(args.Service.Registries) && args.Config.FileDir == "" {
-		client, kuberr := kubelib.CreateClientset(args.Config.KubeConfig, "")
+	if hasKubeRegistry(args.Service.Registries) {
+		// We will also need the rest config - where the public key of k8s is stored - use the new method.
+		kc := args.Config.KubeConfig
+		if kc == "" {
+			kc = os.Getenv("KUBECONFIG")
+		}
+		kcfg, kuberr := kubelib.BuildClientConfig(kc, "")
+		if kuberr != nil {
```

Comment:
You changed the kubeconfig loading orders: previously 

	//Config loading rules:
	// 1. kubeconfig if it not empty string
	// 2. In cluster config if running in-cluster
	// 3. Config(s) in KUBECONFIG environment variable
	// 4. Use $HOME/.kube/config

Reply:
Yes, it is intentional - any reason not to ? 
I'm trying to get rid of CLI flags (so it's easier to start) - keeping kubeconfig for backward compat but istiod standalone won't use it. 

And if running in-cluster - if KUBECONFIG is set it will use that, since it's an explicit configuration.

Reply:
Reverted it - to be extra safe. I can't imagine any side effect, but we can do it later.

Reply:
I agree we can file an issue and discuss later